### DISCUSSION TITLE
Welcome to ESP-Google-Sheet-Client Discussions!

### DISCUSSION THREAD

**Discussion by mobizt** [mobizt/ESP-Google-Sheet-Client#1] [Announcements]
## 👋 Welcome!
  This Discussions is for the library Q & A.

**Comment by dzario** [comment]
Hi.
I use you lib, work .. but I don't know every time bool result is true .
For example : 
` bool success = GSheet.values.append(&response /* returned response */ ...`
return still true .. but data not append do file .. if reset it then start working again.. 
How I can check so data are write to my sheet ?
`response.toString(Serial, true);`
test if respond have any text ?

regards ..

**Reply by mobizt** [reply]
You should check and parse the value manually from the server response from  `response.toString(Serial, true);`.

You also should read the Google Sheet REST API documentation.

I can't provide any method to verify the process unless show what's the server returning.

**Reply by dzario** [reply]
Thanks for fast respond.
Yes , right now I test it .. very easy test if respond strlen of respond is more then 250 character .. then all si OK .
But I see this if get any error then respond is : 
```
{
	"error": {
		"code": 404,
		"message": "Requested entity was not found.",
		"status": "NOT_FOUND"
	}
}
```
There is bad Spread Sheet ID 
and if all is OK then get : 
```
{
	"spreadsheetId": "1NO1FFWAwbXCryi9bcbD1h99oszagPmJ7g2S3pSWSTWI",
	"tableRange": "NewSheet!A1:D19257",
	"updates": {
		"spreadsheetId": "1NO1FFWAwbXCryi9bSbD1h99oszagPmJ7g2S3pSWSTWI",
		"updatedRange": "NewSheet!A19258:D19258",
		"updatedRows": 1,
		"updatedColumns": 4,
		"updatedCells": 4
	}
}
```
mean all is OK .. 
Now .. how easy check if in respond is error or not ..
Is possible use : strstr function or is better .. ?

**Reply by mobizt** [reply]
For the FirebaseJson usage, see [this example](https://github.com/mobizt/ESP-Google-Sheet-Client/blob/master/examples/FirebaseJson/BasicUsage/Create_Edit_Parse/Create_Edit_Parse.ino).

**Reply by dzario** [reply]
Yes, you write some very useful example and I use it .
But there another question ..
If create New Spread Sheet is possible define format for column (number,text, datetime )? color , dimension cell (column) , font ?

thanks .

**Reply by mobizt** [reply]
The library is the REST client for Google Sheet API then you should read the documentation provided by Google.

**Reply by dzario** [reply]
Thanks , I try read , if understand :) .
Thanks for all..

**Reply by dzario** [reply]
Ehm, is possible help me to understand something ..
I create Service Account .. and use to communicate with my Sheet.\
But what is `#define CLIENT_EMAIL "esp32-691@esp32-348114.iam.gserviceaccount.com"` who is owner, what maximum space is for this account ? Where I can more read about it ?

**Reply by mobizt** [reply]
The account of Google cloud applications and services for authentication to use or access the project resources only.
Not a user account that owned everything.

https://cloud.google.com/iam/docs/service-accounts

**Reply by dzario** [reply]
if understand then CLIENT_EMAIL is only virtual user, data are real stored in my account , I must translate , because my English not very good . Your code work very good. 

Thanks for your time . 
Best regards.

**Reply by mobizt** [reply]
Not an any kind of user but it is special account that acts as admin that can control all application access and resources.

**Reply by dzario** [reply]
Thanks.
